### PR TITLE
feat(infra): introduce nextjs

### PR DESCRIPTION
**Pull request by AugustinMauroy**
## Description

This PR introduce basic of the project.

Done:

- adding nextjs
- adding storybook
- adding formating
  - Prettier as formater
  - ES-Lint as js linter
  - StyleLint as style linter
- Use turborepo for caching and perf
- use `.nvmrc` to say which node version we are using.
- setup tailwindcss with config from https://etherpad.org

Why use `npm` ? Because it's the most used package manager. Source [nodejs/next-10](https://github.com/nodejs/next-10/blob/main/surveys/2023-04/)
Why propose vscode extension ? Because it's one of most used IDE so we can be sure that the contributors are in good condition to make superb features.

## Related Issues

No related Issues.

### Check List

- [X] I have formatted using `npx turbo format`
- [X] I have checked the types using `npx turbo type-check`
- [X] I have tested the build using `npx turbo build`

### CODE REVIEW

**Comment by AugustinMauroy** [comment]
BTW: with nextjs squash commit is more sexy so when you feel confortable with this pr use squash and merge

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+version: 2
+updates:
+  - package-ecosystem: github-actions
+    directory: '/'
+    schedule:
+      interval: monthly
```
Comment (SamTV12345) [inline .github/dependabot.yml]: I think weekly would be better. Otherwise we get a ton of prs at the end of the month.

**Review by SamTV12345 — COMMENTED** [review]
Maybe switching to pnpm would be better. We use pnpm also in the repository.

File: .github/dependabot.yml (hunk 1 above)
Comment (SamTV12345) [reply .github/dependabot.yml]: Other than that great job.

**Comment by AugustinMauroy** [comment]
> Maybe switching to pnpm would be better. We use pnpm also in the repository.

in pr description i had explain why I have selected npm but you can give an other opinion

**Comment by SamTV12345** [comment]
> > Maybe switching to pnpm would be better. We use pnpm also in the repository.
> 
> in pr description i had explain why I have selected npm but you can give an other opinion

Ok understood. We can first go with npm. Pnpm is a wrapper around npm and simplifies detecting errors in package.json and also gets rid of npm install --legacy-peer-deps because something is not configured correctly.

File: .github/dependabot.yml (hunk 1 above)
Comment (AugustinMauroy) [reply .github/dependabot.yml]: it's true that we're on a webapp and we need to react to dependencies containing a Security fail.

Comment (SamTV12345) [reply .github/dependabot.yml]: Yes. That is indeed quite important. But with dependabot it's a piece of cake.

**Comment by AugustinMauroy** [comment]
I think we're ready to merge. It's time to say that the project is officially launched 🎉🥳

**Comment by SamTV12345** [comment]
yaaay. Awesome

**Pull request merged**